web/src/components/BlockList.vue

```diff
@@ -0,0 +1,293 @@
+        fixed-header
+        height="71vh"
+        hide-default-footer
+        id="blockList"
+        item-key="id"
+        multi-sort
+        selectable-key="active"
+        single-select
+        style="width: 100%"
+        v-model="dataTableSelection"
+      >
+        <template v-slot:item.active="{ item }">
+          <v-simple-checkbox
+            :ripple="false"
+            @input="dataTableCheckBoxToggle(item, $event)"
+            color="primary"
+            off-icon="mdi-eye-off"
+            on-icon="mdi-eye"
+            v-model="item.active"
+          ></v-simple-checkbox>
+        </template>
+      </v-data-table>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import BlockVisualizer from "@/components/BlockVisualizer.vue";
+import { Block, Diff } from "@/api/api";
+import { Component, Prop, PropSync, Vue, Watch } from "vue-property-decorator";
+import { constructID, SelectionId } from "@/util/OccurenceHighlight";
+import { SideID } from "@/components/CompareCard.vue";
+
+interface BlockWithId extends Block {
+  id: number;
+}
+
+@Component({
+  components: { BlockVisualizer },
+  methods: {
+    constructID
+  }
+})
+export default class BlockList extends Vue {
+  @Prop({ required: true }) diff!: Diff;
+  @Prop({ required: true }) selected!: {
+    side: SideID;
+    blockClasses: Array<SelectionId>;
+  };
+
+  @PropSync("selectedItemSync", { required: true }) selectedItem!: number;
+
+  headers = [
+    {
+      text: "visibility",
+      sortable: true,
+      value: "active"
+    },
+    {
+      text: "K-mers",
+      sortable: true,
+      value: "pairs.length"
+    }
+  ]
+
+  selectionsIds!: Array<[SelectionId, SelectionId]>;
+  dataTableSelection: [BlockWithId] | [] = [];
+
+  get blockLengths(): Array<number> {
+    return this.diff.blocks?.map(block => block.pairs.length)!;
+  }
+
+  get lowestBlockLength(): number {
+    return this.blockLengths.reduce((pv, cv) => Math.min(pv, cv)) as number;
+  }
+
+  get highestBlockLength(): number {
+    return this.blockLengths.reduce((pv, cv) => Math.max(pv, cv)) as number;
+  }
+
+  get selectedBlock(): Block | undefined {
+    return this.diff.blocks![this.selectedItem];
+  }
+
+  get anyActive(): boolean {
+    return this.diff.blocks?.some(block => block.active) as boolean;
+  }
+
+  get blocksWithId(): Array<BlockWithId> {
+    return this.diff.blocks!.map((block, index) => {
+      const blockWithId = (block as BlockWithId);
+      blockWithId.id = index;
+      return blockWithId;
+    });
+  }
+
+  applyMinBlockLength(value: number): void {
+    for (const block of this.diff.blocks!) {
+      block.active = value <= block.pairs.length;
+    }
+    if (this.selectedBlock && !this.selectedBlock.active) {
+      this.selectedItem = -1;
+    }
+  }
+
+  changeSelectedItem(dx: number, current?: number): void {
+    if (!this.anyActive) {
+      this.selectedItem = -1;
+    } else {
+      const length = this.diff.blocks!.length;
```

Comment:
I would add an explicit assertion to check if something is not undefined. This will also tell typescript it can't be undefined after this.